### PR TITLE
Fix declaration warnings and enable java in no_extra_output tests.

### PR DESCRIPTION
**Pull request by aardvark179**

### CODE REVIEW

File: lib/truffle/java/env_java.rb [hunk 1]
```diff
@@ -37,4 +37,10 @@ def inspect
...
 end
 
-ENV_JAVA = Java::SystemPropertyMap.new
+verbose = $VERBOSE
+begin
+  $VERBOSE = nil
+  ENV_JAVA = Java::SystemPropertyMap.new
+ensure
+  $VERBOSE = verbose
+end
```
Comment (eregon) [inline lib/truffle/java/env_java.rb]: Why do we need ENV_JAVA defined before this? There are some gems using it but not requiring `java`?
Maybe one way to fix this would be to define ENV_JAVA in core, and just use a couple primitives.

Another way to avoid the warning would be to `remove_const :ENV_JAVA`.
But this seems fine for now.

File: lib/truffle/java/java.rb [hunk 2]
```diff
@@ -147,9 +147,6 @@ def method_missing(name, *args)
       val = JavaUtilities.get_relative_package_or_class(self, name)
       @children[name] = val
 
-      if name[0] == name[0].upcase && !val.kind_of?(JavaPackage)
-        const_set(name, val)
-      end
```
Comment (eregon) [inline lib/truffle/java/java.rb]: Is this change intentional?

Comment (aardvark179) [reply lib/truffle/java/java.rb]: Yes. It turns out constants was being set in two places when resolving a Java class so each lookup produced a warning.
